feat: refactor filtering - advanced searching

Nerivec opened the pull request:
* feat: unified filtering across the board (interface/logic)
* feat: add global filtering (single input that looks across columns/datapoints)
* feat: better advanced searches
  * by text, by range, by select, by checkbox
  * select and text with auto-complete (suggestions) based on "all present values" (not relying on static/incomplete lists)
  * all filters stored in browser local storage, persisting across page close/reload (_except private windows & cache clearing_)
  * can search even hidden columns
  * beyond filtering: can be used to quickly identify min/max (in range placeholders) & counts (in select & text suggestions parentheses) for specific columns
* feat: add ability to hide static info alerts throughout the UI with a setting (saved in browser local storage)
* fix: Dashboard now uses react-table (headless) to handle the data to match filtering capabilities & co with tables (while keeping the masonry layout)
* fix: switch to built-in row selection for OTA page to cleanup some of the logic
* fix: consolidate store keys for tables
* fix: debounce re-rendering issues
* fix: various styling alignments

@thargy @thk-socal @KoalaWerewolf any chance you can go bug hunting on this? 😉

thargy commented:
I love it! The interface is much cleaner and modern, making it much more accessible for new users in particular. The advanced search is significantly more powerful. The universal search is awesome too.

### Suggestions

1. Universal search searches everything except Last seen (which is fair enough) and Availability. You could include availability, but not a big deal.
2. On dashboard universal search would be great if it searched exposed features and feature type (e.g. 'brightness', 'presence', 'occupancy'). Again not a big deal though.
3.  Groups -> the create group takes up loads of space and isn't obviously a button. In fact on Safari on Mac I had a few random cases of clicking and it not expanding. I think it'd make more sense as an actual button, next to advanced search that opens a modal like advance search.
4. I still think that persisting the filter across pages (Devices, Dashboard, OTA) would be ideal. Still, given the different data loaded and potential performance impact, it's a reasonable compromise to remember filters per page as you've done.

Again, this is great as is, and a big improvement, meaning the next release will not only be game-changing for multi-connection systems, but even for single-system users.

Nerivec commented:
@thargy 
1. I've limited some searches depending on pages to avoid having too many "why is that matching?". About last seen, it's indeed rather complicated to search (number, text, date, etc.), even in advanced search I added a small tip to make it clear it's minutes. Availability is "available" on Devices page, but that stuff is not loaded in dashboard (because not shown/used).
2. That would likely render the global search unhelpful for a lot of users (far too many matches that would conflict with friendly names, at least for EN users).
3. Removed the collapsible for now (takes as much space as the content anyway)
4. Best to keep it per-page I think, probably several users that will keep specific filters "always like this" on a page, and then differently on another.

@KoalaWerewolf 
About your language, I guess you had it set in the browser storage, but before it wasn't available, so it defaulted back to EN. And yes, the "AI" is because it was AI-generated and has yet to be [peer-reviewed](https://github.com/Nerivec/zigbee2mqtt-windfront/issues/199) (I don't speak that many languages, had to put a warning about it 😬).

1. I'm not seeing this, can you reproduce on any browser? _Or maybe I'm not understanding what your meant properly._
2. Not seeing this either. Although this could be something about the device's own "stretch" behavior when hitting top/bot. I'll keep it in my todo list, let's see if it comes up for other users.
3. 👍 I'll change it so it only shows up as `error` when it actually fails (all 10 attempts), else it will log as `debug` in Logs page only

KoalaWerewolf commented:
The column resizing on iPhone, HA Add-on:

https://github.com/user-attachments/assets/815b9882-0cde-412b-8827-b40748048b67

KoalaWerewolf commented:
Since I mostly use the HA add-on I've spun up my docker instance as well to test there. No difference, tried multiple browsers on my iPhone. Not sure how much difference the browser makes on iOS, don't they all still use WebKit?

Nerivec commented:
Browser engines are [far messier than that](https://www.geeksforgeeks.org/techtips/web-browser-engine-definition-working/).
I tried both Firefox and Chrome on Android, can't reproduce on either. No Apple stuff here though, so can't test that.
If you have some time to kill, see if maybe it is related to one of your devices that may have a combination of parameters I'm missing to properly reproduce -this stuff is all about text length, column width, etc.- (will have to filter in batches to see if omitting a device eventually makes this go away).

KoalaWerewolf commented:
I've tried different filters, a single device with a short name still behaves the same way. With a single device filtered there is nothing to scroll, so the table width never adjusts, a reload is required for the table to update.

The browser engine topic on iOS is different than other platforms afaik. For a long time a different browser still meant the same engine, looks like that's still true. 
https://www.howtogeek.com/iphone-missing-custom-rendering-engine-browsers/
https://appleinsider.com/articles/25/07/14/apple-still-effectively-blocks-rival-browser-engines-on-ios-despite-eu-order

Nerivec commented:
My bad, I thought that stuff was already in effect... (_should have known better_)
I don't have any way to test this though atm.
Have you tried with iOS 26 by any chance? (I know they had introduced some bugs in that version for Safari, but hopefully fixed now...).
Did you try with something else than Apple to confirm where the issue is?
